python/torch_mlir_e2e_test/torchscript/framework.py

```diff
@@ -238,40 +269,89 @@ def generate_golden_trace(test: Test) -> Trace:
-                           compilation_error=None,
-                           runtime_error="".join(traceback.format_exception(
-                               type(e), e, e.__traceback__)),
-                           trace=None,
-                           golden_trace=None))
-            continue
+def compile_and_run_test(test_name: str, results: mp.Queue, tests: Dict[str, Test],
+                 config: TestConfig) -> Any:
+    test = tests[test_name]
 
+    try:
+        golden_trace = generate_golden_trace(test)
+        compiled = config.compile(test.program_factory())
+    except Exception as e:
         results.append(
             TestResult(unique_name=test.unique_name,
-                       compilation_error=None,
+                       compilation_error="".join(
+                           traceback.format_exception(type(e), e,
+                                                      e.__traceback__)),
                        runtime_error=None,
-                       trace=trace,
-                       golden_trace=golden_trace))
+                       trace=None,
+                       golden_trace=None))
+        return
+    try:
+        trace = config.run(compiled, golden_trace)
+    except Exception as e:
+        results.append(
+            TestResult(unique_name=test.unique_name,
+                       compilation_error=None,
+                       runtime_error="".join(
+                           traceback.format_exception(type(e), e,
+                                                      e.__traceback__)),
+                       trace=None,
+                       golden_trace=None))
+        return
+
+    results.append(TestResult(unique_name=test.unique_name,
+                        compilation_error=None,
+                        runtime_error=None,
+                        trace=clone_trace(trace),
+                        golden_trace=golden_trace))
+
+
+def run_tasks_in_parrallel(task_queue: mp.Queue, task):
+    NUMBER_OF_PROCESSES = min(mp.cpu_count(), task_queue.qsize())
+    processes = []
+    for i in range(NUMBER_OF_PROCESSES):
+        p = mp.Process(target=task, args=(task_queue, ))
+        p.start()
+        processes.append(p)
+    for i in range(NUMBER_OF_PROCESSES):
+        task_queue.put('STOP')
+    for p in processes:
+        p.join()
+
+
+def run_tests(tests: List[Test], config: TestConfig) -> List[TestResult]:
+    """Invoke the given `Test`'s with the provided `TestConfig`."""
+
+    manager = mp.Manager()
+    tests_queue = manager.Queue()
+    task_queue = manager.Queue()
+    sync_results = manager.list()
+    # This is needed because autograd does not support crossing process
+    # boundaries.
+    torch.autograd.set_grad_enabled(False)
+    for test in tests:
+        tests_queue.put(test.unique_name)
+
+    test_dict = {test.unique_name: test for test in tests}
+
+    def worker(q: mp.Queue):
+        for test_name in iter(q.get, 'STOP'):
```

Comment:
nit: `"QUEUE_SENTINEL"` to make it more obvious (also, for consistency, use `"`)